### ISSUE TITLE
Sorting NULL first/last

### ISSUE THREAD

**Issue by JM-Agrimap** [realm/realm-java#3209]
When sorting it would be nice to be able to specify if NULL values in the column should be at the beginning or end of the results.

The current behaviour is for NULLs to be at the beginning of the results if ASC and end if DESC.

In my application I am querying a list of records by "date due descending". 
The requirement for the UI is to display records without a due date first followed by those with a due date. Currently all those without a due date appear at the bottom of the list.

To make it easy to use I would just suggest enhancing the io.realm.Sort enum by adding ASCENDING_NULLS_FIRST/LAST and DESCENDING_NULLS_FIRST/LAST.

**Comment by Zhuinden** [comment]
In the meantime, you can construct your displayed elements from multiple RealmResults.

http://stackoverflow.com/a/34976540/2413303

**Comment by stk1m1** [comment]
@JM-Agrimap 

It saddens me to tell you the feature you've sought after is not currently supported. I'll place this suggestion in `enhancement` to track in the future. Thanks!

**Comment by stk1m1** [comment]
**Internal Note**

It looks like this issue has been discussed about an year ago here <https://github.com/realm/realm-java/issues/3209> but revoked due to be aligned with other databases.

**Comment by Zhuinden** [comment]
@stk1m1  technically Realm-Java started growing on its own in the `Android` section as its own topic:

http://stackoverflow.com/documentation/android/3187/realm-io

**Comment by stk1m1** [comment]
@Zhuinden 

Thanks for the link. 😉 Btw, I think you'd rather want to put the link in #3202 so we can collect related info all in one place. I'll remove my comment as soon as you do that.

**Comment by mussa-ibragimov** [comment]
It's almost 2019 :(

**Comment by Zhuinden** [comment]
And you're pretty much the second person to ask for it so far, so don't keep your hopes up.

**Comment by PacoSwifty** [comment]
Well I'll be the fourth; I'm currently in need of exactly this capability.

**Comment by mussa-ibragimov** [comment]
I'm sure the number is a lot more than four. Everybody is just hacking it with multiple RealmResults.

**Comment by chrisbln** [comment]
You can solve the issue with two fields in our object instead of using multiple lists, which is much easier if you can update both fields simultaneously.

- For example one for your date and one if it is available
```
class RealmMediaWrapper {
    val releaseDate: Long
    val hasReleaseDate: Boolean
}
```

- Sort first whether the date is available and then the date itself. For example:
```
if(sort == Sort.ASCENDING)
   results.sort(RealmConstant.HAS_RELEASE_DATE, Sort.DESCENDING, RealmConstant.RELEASE_DATE, sort)
else
   results.sort(RealmConstant.RELEASE_DATE, sort)
```

- Migrate old objects if necessary
```
transform {
   val value = it.getLong("releaseDate")
   it.setBoolean("hasReleaseDate", value != 0L)
}
```

**Comment by matthewkrueger** [comment]
Another vote for this. Will use the hacky solution for now.

**Comment by matthewkrueger** [comment]
Thanks for the trick @chrisbln - just tried it out and it's working great for now. A little annoying to add an extra field for this, but sorting in realm is so much faster than trying to sort in memory.

**Comment by whalemare** [comment]
Bump. Would be nice to have this feature.
For now we should create extra sorting fields or concat multiple lists

**Comment by gustavomts** [comment]
Yeah, having that feature would be great because sorting in memory is way slower.
